aten/src/ATen/native/TensorCompare.cpp

```diff
@@ -217,7 +217,7 @@ Tensor where(const Tensor& condition, const Tensor& self, const Tensor& other) {
               condition.device(), " and x and y are on ", self.device(), " and ", other.device(),
               " respectively");
   TORCH_CHECK(condition.scalar_type() == ScalarType::Byte || condition.scalar_type() == ScalarType::Bool,
-              "Expected condition to have ScalarType Byte, but got ScalarType ",
```

Comment:
Let's tweak this a bit

```
if (condition.scalar_type() == ScalarType::Byte) {
  TORCH_WARN_ONCE("where received a uint8 condition tensor. This behavior is deprecated and will be removed in a future version of PyTorch. Use a boolean condition instead.");
} else {
  TORCH_CHECK(condition.scalar_type() == ScalarType::Bool, "where expected condition to be a boolean tensor, but got a tensor with dtype ", condition.scalar_type());
}
```

That updates the error's language and correctly warns when the condition is a byte tensor.